src/main/java/com/arpnetworking/clusteraggregator/client/HttpSourceActor.java

```diff
@@ -0,0 +1,327 @@
+
+            return FlowShape.of(split.in(), createRequest.out());
+        });
+    }
+
+    @Override
+    public Receive createReceive() {
+        return receiveBuilder()
+                .match(HttpRequest.class, request -> {
+                    final ActorRef sender = sender();
+                    Source.single(request)
+                            .via(_processGraph)
+                            .toMat(_sink, Keep.right())
+                            .run(_materializer)
+                            .whenComplete((done, err) -> {
+                                if (err == null) {
+                                    sender.tell(HttpResponse.create().withStatus(200), self());
+                                } else {
+                                    if (err instanceof NoRecordsException) {
+                                        sender.tell(HttpResponse.create().withStatus(400), self());
+                                    } else {
+                                        BAD_REQUEST_LOGGER.warn()
+                                                .setMessage("Error handling http post")
+                                                .setThrowable(err)
+                                                .log();
+                                        sender.tell(HttpResponse.create().withStatus(500), self());
+                                    }
+                                }
+                            });
+                })
+                .build();
+    }
+
+    private static ImmutableMultimap<String, String> createHeaderMultimap(final Iterable<HttpHeader> headers) {
+        final ImmutableMultimap.Builder<String, String> headersBuilder = ImmutableMultimap.builder();
+
+        for (final HttpHeader httpHeader : headers) {
+            headersBuilder.put(httpHeader.lowercaseName(), httpHeader.value());
+        }
+
+        return headersBuilder.build();
+    }
+
+    private static com.arpnetworking.clusteraggregator.models.HttpRequest mapModel(
+            final Pair<ByteString, ImmutableMultimap<String, String>> pair) {
+        return new com.arpnetworking.clusteraggregator.models.HttpRequest(pair.second(), pair.first());
+    }
+
+    private List<AggregationMessage> parseRecords(final com.arpnetworking.clusteraggregator.models.HttpRequest request) {
+        final ArrayList<AggregationMessage> records = Lists.newArrayList();
+        ByteString current = request.getBody();
+        Optional<AggregationMessage> messageOptional = AggregationMessage.deserialize(current);
+        while (messageOptional.isPresent()) {
+            final AggregationMessage message = messageOptional.get();
+            records.add(message);
+            current = current.drop(message.getLength());
+            messageOptional = AggregationMessage.deserialize(current);
+            if (!messageOptional.isPresent() && current.lengthCompare(4) > 0) {
+                LOGGER.debug()
```

Comment:
Under TCP this is probably ok, but under HTTP this would indicate a bad request right? Should we discard everything? Return non-200? Or just log a warning?

Reply:
^ Clients are potentially losing data.